### PR TITLE
feat: invite user to answer question

### PR DESCRIPTION
**Pull request by HuanCheng65**
## Summary by CodeRabbit

- **New Features**
	- Introduced new data models and relations for handling question invitations and user interactions.
	- Added new DTO (Data Transfer Object) classes for managing question invitations, user answers, and question recommendations.
	- Implemented new functionalities in the questions service, including inviting users to answer questions, fetching invitation details, and recommending questions.
	- Enhanced user service with authorization configurations for resource management.
- **Bug Fixes**
	- Addressed minor issues in test configurations and corrected a typo in function names.
- **Chores**
	- Updated `.gitignore` to exclude specific avatar images.
- **Documentation**
	- Improved code documentation and structure for better readability and maintenance.

### CODE REVIEW

File: src/questions/questions.service.ts [hunk 1]
```diff
@@ -527,6 +540,149 @@ export class QuestionsService {
...
+    const Invitation =
+      await this.prismaService.questionInvitationRelation.create({
+        data: {
+          questionId,
+          userId,
+        },
+      });
+    return {
+      userId,
+      success: true,
```
Comment (andylizf) [inline src/questions/questions.service.ts:573]: what's this?

File: src/questions/questions.controller.ts [hunk 2]
```diff
@@ -274,4 +281,102 @@ export class QuestionsController {
       },
     };
   }
+
+  @Get('/:id/invitations')
```
Comment (andylizf) [inline src/questions/questions.controller.ts:285]: You can create a new file and seperate these functions into QuestionsController in that file utilizing the partial definition of Typescript.

File: test/question.e2e-spec.ts [hunk 3]
```diff
@@ -607,6 +613,181 @@ describe('Questions Module', () => {
...
+        .post(`/questions/${questionIds[1]}/invitations`)
+        .set('Authorization', `Bearer ${TestToken}`)
+        .send(userIds);
+      expect(respond.body.message).toBe('Invited');
+      expect(respond.body.code).toBe(201);
+      expect(respond.status).toBe(201);
+      expect(respond.body.data.userId).toEqual(TestUserId);
+      invitaionIds.push(respond.body.data.invitationId);
+    });
+    it('should invite some users to answer question', async () => {
```
Comment (andylizf) [inline test/question.e2e-spec.ts:629]: different description plz.

**Review by andylizf — CHANGES_REQUESTED** [review]

File: src/questions/questions.service.ts [hunk 4]
```diff
@@ -53,6 +61,8 @@ export class QuestionsService {
     private readonly questionFollowRelationRepository: Repository<QuestionFollowerRelation>,
     @InjectRepository(QuestionSearchLog)
     private readonly questionSearchLogRepository: Repository<QuestionSearchLog>,
+    @InjectRepository(QuestionInvitation)
+    private readonly questionInvitationRepository: Repository<QuestionInvitation>,
```
Comment (andylizf) [inline src/questions/questions.service.ts:65]: used?

File: src/questions/questions.service.ts [hunk 5]
```diff
@@ -527,6 +540,154 @@ export class QuestionsService {
...
+    page_start: number;
+    has_prev: boolean;
+    has_more: boolean;
+  }> {
+    const orderField = sort === '+createdAt' ? 'asc' : 'desc';
+    const questionDto = await this.getQuestionDto(questionId);
+    const questionInvitations =
+      await this.prismaService.questionInvitationRelation.findMany({
+        where: { questionId },
+        orderBy: { createdAt: orderField }, // 使用正确的字段名
```
Comment (andylizf) [inline src/questions/questions.service.ts:593]: ?

File: test/question.e2e-spec.ts [hunk 6]
```diff
@@ -10,6 +10,7 @@ import { INestApplication, Logger } from '@nestjs/common';
 import { Test, TestingModule } from '@nestjs/testing';
 import request from 'supertest';
 import { AppModule } from '../src/app.module';
+import { sendIdDto } from '../src/questions/DTO/send-id.dto';
```
Comment (andylizf) [inline test/question.e2e-spec.ts:13]: UpperCamelCase

File: src/questions/questions.controller.ts [hunk 7]
```diff
@@ -274,4 +281,109 @@ export class QuestionsController {
...
+    };
+  }
+
+  @Delete('/:id/invitations')
+  async cancelInvition(
+    @Param('id', ParseIntPipe) id: number,
+    @Headers('Authorization') auth: string | undefined,
+    @Body() InvitationId: sendIdDto,
+  ): Promise<cancelInvitationResponseDto> {
+    const userId = this.authService.verify(auth).userId;
```
Comment (andylizf) [inline src/questions/questions.controller.ts:346]: audit

Comment (andylizf) [reply src/questions/questions.controller.ts:346]: invitations related all need audit

**Review by andylizf — CHANGES_REQUESTED** [review]

**Comment by Nictheboy** [comment]
@coderabbitai

**Comment by Nictheboy** [comment]
@coderabbitai review

**Review by andylizf — APPROVED** [review]

File: src/questions/questions.controller.ts [hunk 8]
```diff
@@ -274,4 +279,119 @@ export class QuestionsController {
...
+        page,
+      },
+    };
+  }
+
+  @Post('/:id/invitations')
+  async inviteUserAnswerQuestion(
+    @Param('id', ParseIntPipe) id: number,
+    @Headers('Authorization') auth: string | undefined,
+    @Body('id', ParseIntPipe) invitedUserId: number,
```
Comment (HuanCheng65) [inline src/questions/questions.controller.ts:315]: It should be `user_id`

File: src/questions/questions.controller.ts [hunk 9]
```diff
@@ -274,4 +279,119 @@ export class QuestionsController {
...
+
+  @Get('/:id/invitations')
+  async getQuestionInvitations(
+    @Param('id', ParseIntPipe) id: number,
+    @Query('page_start', new ParseIntPipe({ optional: true }))
+    pageStart: number | undefined,
+    @Query('page_size', new ParseIntPipe({ optional: true }))
+    pageSize: number | undefined,
+    @Query('sort', new ParseEnumPipe({ optional: true }))
+    sort: '+createdAt' | '-createdAt',
```
Comment (HuanCheng65) [inline src/questions/questions.controller.ts:291]: ???????????????????????????????????????????????????????????????????????????????????????????????

File: src/questions/questions.controller.ts [hunk 10]
```diff
@@ -274,4 +279,119 @@ export class QuestionsController {
...
+    this.authService.audit(
+      auth,
+      AuthorizedAction.delete,
+      await this.questionsService.getInvitedById(id, invitation_id),
+      'questions/invitation',
+      invitation_id,
+    );
+    await this.questionsService.cancelInvitation(id, invitation_id);
+    return {
+      code: 204,
```
Comment (HuanCheng65) [inline src/questions/questions.controller.ts:353]: Why not respond with `204` directly?

**Pull request merged**